### PR TITLE
Add support for pyarrow 6

### PR DESCRIPTION
**Pull request by ldacey**
Addresses #324

### CODE REVIEW

**Comment by ldacey** [comment]
@xhochy Is there a way for me to run these checks again without committing some unnecessary change? I see that pyarrow 6 is on conda-forge now so I think my original pull request should be successful.

**Comment by xhochy** [comment]
Closing/opening a PR retriggers the tests ;)

File: Earthfile [hunk 1]
```diff
@@ -239,16 +248,25 @@ test-python3.9-arrow4.x.x:
...
         --build-arg ARROW_VERSION_RULE=">=5,<6" \
         --build-arg NUMPY_VERSION_RULE=">=1.20.0" \
         +test/result /result
 
     SAVE ARTIFACT /result AS LOCAL result
 
+test-python3.9-arrow6.x.x:
+    ARG PYTHON_VERSION="3.9.7"
+    COPY --build-arg PYTHON_VERSION="$PYTHON_VERSION" \
+        --build-arg ARROW_VERSION_RULE=">=7,<7" \
```
Comment (xhochy) [inline Earthfile:262]:
```suggestion
        --build-arg ARROW_VERSION_RULE=">=6,<7" \
```

**Comment by xhochy** [comment]
Seems like we have an actual error:

```
              +build | /src/cpp/turbodbc_arrow/Library/src/set_arrow_parameters.cpp: In function 'std::shared_ptr<arrow::Table> turbodbc_arrow::unwrap_pyarrow_table(const pybind11::object&)':
              +build | /src/cpp/turbodbc_arrow/Library/src/set_arrow_parameters.cpp:427:64: error: too many arguments to function 'arrow::Result<std::shared_ptr<arrow::Table> > arrow::py::unwrap_table(PyObject*)'
              +build |   427 |     if (not arrow::py::unwrap_table(pyarrow_table.ptr(), &table).ok()) {
              +build |       |              
```

**Comment by pacman82** [comment]
That might very well be related to #317  .

**Comment by ldacey** [comment]
> Closing/opening a PR retriggers the tests ;)

Ah nice thanks, I'll keep that in mind for next time. 

Regarding the error, I do not know C++ at all. The message above seemed to be about the second argument to this function: `arrow::py::unwrap_table(pyarrow_table.ptr(), &table)`, which is what @jorisvandenbossche mentioned in #317.

In the code from pyarrow.cc, it looks like `unwrap_##FUNC_SUFFIX(PyObject* obj)` only accepts the object argument which in the turbodbc code seems to be `pyarrow_table.ptr()`. I thought perhaps `std::shared_ptr<TYPE_NAME>* out` (which has been removed) is the `&table` argument. So perhaps simply removing that would work.

I tried to remove `&table` locally from that function so it only has one argument ran the earthly `test-python3.9-arrow6.x.x` test.

First three tests passed quickly:
```
+test | 1/4 Test #1: cpp_odbc_unit_test ...............   Passed    0.05 sec
+test | 2/4 Test #2: turbodbc_unit_test ...............   Passed    0.11 sec
+test | 3/4 Test #3: turbodbc_arrow_unit_test .........   Passed    0.03 sec
```

Final test took a bit longer to complete, but ultimately failed. Segmentation fault during the `test_executemanycolumns` test. Not sure if it is caused by the change I made.


```

+test |     Start 4: turbodbc_integration_test
+test *failed* | 4: ============================= test session starts ==============================
+test *failed* | 4: platform linux -- Python 3.9.7, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
+test *failed* | 4: rootdir: /src
+test *failed* | 4: plugins: cov-3.0.0
+test *failed* | 4: collected 420 items
+test *failed* | 4: 
+test *failed* | 4: test_connect.py ........
+test *failed* | 4: test_connection.py ............
+test *failed* | 4: test_constructors.py ...
+test *failed* | 4: test_cursor_async_io.py .
+test *failed* | 4: test_cursor_basics.py ............FF
+test *failed* | 4: test_cursor_insert.py ..FF.FF.FF.FF.FF.FF.FF.FF..FF.FF.FF.FF.FF.FF.FF.FF.FF.FF.FF.FF
+test *failed* | 4: test_cursor_select.py .FF.FF.FF.FF.FF.FF.FF.FF.FF.FF.FF.FF.FF.FF.FF.FF.FF.FF.FF.FF.FF.FF.F..
+test *failed* | 4: test_cursor_unsupported_features.py ..
+test *failed* | 4: test_data_types.py ..
+test *failed* | 4: test_executemanycolumns.py sFatal Python error: Segmentation fault
+test *failed* | 4: 
+test *failed* | 4: Current thread 0x00007ff6a8fba740 (most recent call first):
+test *failed* | 4:   File "/src/build/turbodbc/cursor.py", line 201 in executemanycolumns
+test *failed* | 4:   File "/src/build/turbodbc/exceptions.py", line 41 in wrapper
+test *failed* | 4:   File "/src/python/turbodbc_test/test_executemanycolumns.py", line 45 in _test_basic_column
+test *failed* | 4:   File "/src/python/turbodbc_test/test_executemanycolumns.py", line 122 in _full_column_tests
+test *failed* | 4:   File "/src/python/turbodbc_test/test_executemanycolumns.py", line 146 in test_integer_column

```

**Comment by alippai** [comment]
Changing the code linked by @pacman82 and @jorisvandenbossche seems to help, however someone check it, since I didn't write C++ in the past 10 years and I literally have no idea how this works. #fixedaccidentally 
https://github.com/blue-yonder/turbodbc/pull/326

**Pull request merged**